lambda_functions/user_service/create_user.py

```diff
@@ -29,13 +29,16 @@ def create_user(event):
                 GroupName = 'Detective'
             )
 
-            # Add mail address if submitted
-            if 'email' in event['request']['userAttributes']:
+            # Add mail address if submitted and set confirmation status
+            if 'custom:mail_subscription' in event['request']['userAttributes']:
```

Comment:
Die if-Abfrage würde ich mit der in Zeile 40 kombinieren - Sonst wird gar kein Mail-Objekt angelegt, wenn dieses Attribut nicht existiert, oder?